pkg/registry/file/vulnsummarystorage.go

```diff
@@ -0,0 +1,240 @@
+	updateVulnCounters(&aggregatedSeverities.Unknown, &severities.Unknown)
+}
+
+func updateFullVulnSumm(fullVulnSumm *softwarecomposition.VulnerabilitySummary, vulnManifestSumm *softwarecomposition.VulnerabilityManifestSummary) {
+	updateSeverities(&fullVulnSumm.Spec.Severities, &vulnManifestSumm.Spec.Severities)
+	if vulnManifestSumm.Spec.Vulnerabilities.ImageVulnerabilitiesObj != (softwarecomposition.VulnerabilitiesObjScope{}) {
+		fullVulnSumm.Spec.WorkloadVulnerabilitiesObj = append(fullVulnSumm.Spec.WorkloadVulnerabilitiesObj, vulnManifestSumm.Spec.Vulnerabilities.ImageVulnerabilitiesObj)
+	}
+	if vulnManifestSumm.Spec.Vulnerabilities.WorkloadVulnerabilitiesObj != (softwarecomposition.VulnerabilitiesObjScope{}) {
+		fullVulnSumm.Spec.WorkloadVulnerabilitiesObj = append(fullVulnSumm.Spec.WorkloadVulnerabilitiesObj, vulnManifestSumm.Spec.Vulnerabilities.WorkloadVulnerabilitiesObj)
+	}
+}
+
+func (s *VulnSummaryStorageImpl) getVulnManifestSummaryDirPath(key string) string {
+	return filepath.Join(s.root, key, "..", "..", "..", "vulnerabilitymanifestsummaries")
+}
+
+// add uni test
+func (s *VulnSummaryStorageImpl) aggregateVulnSummaryOverNamespace(ctx context.Context, fullVulnSumm *softwarecomposition.VulnerabilitySummary, path string) error {
+	var err error
+
+	vulnSumm, err := s.getOneVulnManifestSummary(ctx, path)
+	if err != nil {
+		return err
+	}
+	updateFullVulnSumm(fullVulnSumm, vulnSumm)
+
+	return nil
+}
+
+// add uni test
+func (s *VulnSummaryStorageImpl) aggregateVulnSummary(ctx context.Context, key string) (*softwarecomposition.VulnerabilitySummary, []error) {
+	var errs []error
+	fullVulnSumm := initNamespaceVulnSummary(key)
+
+	namespace := filepath.Base(key)
+	afero.Walk(s.appFs, s.getVulnManifestSummaryDirPath(key), func(path string, info os.FileInfo, err error) error {
+		if namespace != filepath.Base(filepath.Dir(path)) {
+			return nil
+		}
+		if !strings.HasSuffix(path, jsonExt) {
+			return nil
+		}
+		e := s.aggregateVulnSummaryOverNamespace(ctx, &fullVulnSumm, path)
+		if err != nil {
+			errs = append(errs, e)
+		}
+		return nil
+	})
+
+	if len(fullVulnSumm.Spec.WorkloadVulnerabilitiesObj) == 0 {
+		return nil, errs
+	}
+
+	return &fullVulnSumm, errs
+}
+
+// add uni test
+func (s *VulnSummaryStorageImpl) validateKeySupported(ctx context.Context, key string) error {
+	_, span := otel.Tracer("").Start(ctx, "VulnSummaryStorageImpl.Get.validateKeySupported")
+	defer span.End()
+
+	dataFullPath := filepath.Join(s.root, key)
+	if filepath.Base(dataFullPath) == "vulnerabilitysummaries" {
+		return storage.NewMethodNotImplementError(key, "")
+	}
+	return nil
+}
+
+// add uni test
+func (s *VulnSummaryStorageImpl) Get(ctx context.Context, key string, opts storage.GetOptions, objPtr runtime.Object) error {
+	ctx, span := otel.Tracer("").Start(ctx, "VulnSummaryStorageImpl.Get")
+	span.SetAttributes(attribute.String("key", key))
+	defer span.End()
+
+	err := s.validateKeySupported(ctx, key)
+	if err != nil {
+		return err
+	}
+
```

Comment:
thought: this is where we should extract the meaningful data from the key operation

We should extract the kind, namespace and name of the resource that we are requested to process here, and then pass it some business-level object that handles the business logic of building the summary.